### ISSUE TITLE
Vulkan Backend errors when creating a texture array with a layer size of 6.

### ISSUE THREAD

**Issue by StarArawn** [gfx-rs/wgpu#2439]
**Description**
I'm attempting to use texture arrays for my tilemap crate. If I create a texture array with 6 layers the vulkan backend attempts to treat the array as a cube map texture and I get the following error:
```
vkCreateImage(): pCreateInfo->flags contains VK_IMAGE_CREATE_CUBE_COMPATIBLE_BIT, but pCreateInfo->extent.width (=64) and pCreateInfo->extent.height (=32) are not equal. The Vulkan spec states: If imageType is VK_IMAGE_TYPE_2D and flags contains VK_IMAGE_CREATE_CUBE_COMPATIBLE_BIT, extent.width and extent.height must be equal and arrayLayers must be greater than or equal to 6
```

**Repro steps**
Checkout out this repo: https://github.com/StarArawn/bevy_ecs_tilemap/
run: `cargo run --example iso_diamond`

Workaround: Add +1 to count here:
https://github.com/StarArawn/bevy_ecs_tilemap/blob/main/src/render/texture_array_cache.rs#L63
No more errors.

**Expected vs observed behavior**
I expect that my texture array is treated like an array and not a cube map. The texture width/height shouldn't have to match here.

**Extra materials**
n/a

**Platform**
Windows 10, Vulkan, wgpu 0.12.0

**Comment by StarArawn** [comment]
Looking at the code this seems to be the issue here:
https://github.com/gfx-rs/wgpu/blob/0183e7d1e85ac95f2461426a910b2f86f5373119/wgpu-hal/src/vulkan/device.rs#L813
Not sure what a fix would be.

**Comment by kvark** [comment]
It was fixed in #2331 but not backported to crates because it changes the errors, which are public API of wgpu-core.
I think we need to port over a smaller version of this PR that doesn't affect the errors but fixes the problem here.
